lib/global-error/src/macros.rs

```diff
@@ -121,17 +212,90 @@ macro_rules! unwrap_ref {
 }
 pub use unwrap_ref;
 
+/// `AssertionErrorUnwrap` is a trait used for handling unwrapping of `Result`
+/// and `Option` types.
+///
+/// This trait provides a method `assertion_error_unwrap` that takes a `Result`
+/// or `Option` and returns a `Result`. If the original `Result` or `Option` is
+/// `Ok` or `Some`, it returns `Ok` with the unwrapped value. If it's `Err` or
+/// `None`, it returns `Err` with an `AssertionError::Unwrap` that includes a
+/// message and location.
+pub trait AssertionErrorUnwrap {
```

Comment:
I'm fine switching from `AssertionError::Unwrap` to dedicated structures, but it'd be good to standardize it so either it's all one `enum` or each has its own struct (e.g. `AssertionError:Panic` -> `struct AssertionErrorPanic`).

Reply:
I might be misunderstanding, but I think it's all currently in one enum? Or do you mean because of the way I'm handling this right now, it's only every yielding a `AssertionError::Unwrap` when it should also be some others?

Reply:
Solid idea. I inlined it just for testing, but it does take up quite a lot of space.